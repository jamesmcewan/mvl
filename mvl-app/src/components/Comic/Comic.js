import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import Cover from '../Cover/Cover';
import Info from '../Info/Info';

const ComicWrapper = styled.div`
  position: relative;
`;

const Comic = ({ ...rest }) => {
  const [visible, setVisible] = useState(false);

  return (
    <ComicWrapper>
      <Cover {...rest} {...{ visible, setVisible }} />
      <Info {...rest} {...{ visible, setVisible }} />
    </ComicWrapper>
  );
};

Comic.propTypes = {
  title: PropTypes.string,
  thumbnail: PropTypes.object
};

export { ComicWrapper };
export default Comic;
