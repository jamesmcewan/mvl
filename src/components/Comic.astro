---
import Details from './Details.astro'
import type ThumbnailType from '../types/ThumbnailType'

const replaceHttp = (url: string) => url.replace('http', 'https')
const getCover = ({ path, extension }: ThumbnailType) =>
  `${replaceHttp(path)}.${extension}`

const { thumbnail, title, creators, series, prices, pageCount } = Astro.props
---

<div>
  <img
    class="rounded-md"
    src={getCover(thumbnail)}
    width="560"
    height="850"
    alt={title}
  />
  <Details {title} {creators} {series} {prices} {pageCount} />
</div>
