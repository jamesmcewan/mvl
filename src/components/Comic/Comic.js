import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Cover from '../Cover/Cover';
import Info from '../Info/Info';

const ComicWrapper = styled.div`
  position: relative;
  border-radius: 50px;
  overflow: hidden;
`;

const Comic = ({ ...rest }) => {
  const [visible, setVisible] = useState(false);

  return (
    <ComicWrapper>
      <Cover {...rest} {...{ visible, setVisible }} />
      <Info {...rest} {...{ visible, setVisible }} />
    </ComicWrapper>
  );
};

Comic.propTypes = {
  title: PropTypes.string,
  thumbnail: PropTypes.object
};

export { ComicWrapper };
export default Comic;
