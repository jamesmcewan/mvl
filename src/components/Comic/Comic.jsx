import React from 'react';
import PropTypes from 'prop-types';
import Details from '../Details/Details';

const replaceHttp = (url) => url.replace('http', 'https');
const getCover = ({ path, extension }) => `${replaceHttp(path)}.${extension}`;

const Comic = (props) => {
  return (
    <div>
      <img
        className="rounded-md"
        src={getCover(props.thumbnail)}
        width="560"
        height="850"
        alt={props.title}
      />
      <Details {...props} />
    </div>
  );
};

Comic.propTypes = {
  title: PropTypes.string,
  thumbnail: PropTypes.object
};

export default Comic;
