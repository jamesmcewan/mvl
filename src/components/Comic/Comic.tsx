import Details from '@components/Details/Details'
import ComicProps from '../../types/ComicProps'
import ThumbnailType from '../../types/ThumbnailType'

const replaceHttp = (url: string) => url.replace('http', 'https')
const getCover = ({ path, extension }: ThumbnailType) =>
  `${replaceHttp(path)}.${extension}`

const Comic = ({ thumbnail, title, creators }: ComicProps) => {
  return (
    <div>
      <img
        className="rounded-md"
        src={getCover(thumbnail)}
        width="560"
        height="850"
        alt={title}
      />
      <Details title={title} creators={creators} />
    </div>
  )
}

export default Comic
