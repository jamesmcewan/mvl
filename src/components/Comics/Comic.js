import { useState } from 'react'
import Cover from '../Cover/Cover'
import ComicDetails from '../ComicDetails/ComicDetails'

const Comic = ({ ...rest }) => {
  const [isVisible, setIsVisible] = useState(false)

  return (
    <div data-selector="comic-wrapper">
      <Cover {...rest} {...{ isVisible, setIsVisible }} />
      <ComicDetails {...rest} {...{ isVisible, setIsVisible }} />
    </div>
  )
}

export default Comic
