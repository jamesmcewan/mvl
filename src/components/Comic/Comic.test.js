import React from 'react';
import { mount } from 'enzyme';
import Comic from './Comic';
import 'jest-styled-components';

describe('Comic', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Comic />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Comic />);
    expect(wrapper).toMatchSnapshot();
  });
});
