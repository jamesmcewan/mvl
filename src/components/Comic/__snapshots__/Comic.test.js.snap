// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comic should render the correct JSX 1`] = `
.c0 {
  padding: 0;
  margin: 0 3px 3px;
  margin-bottom: 26px;
  border: 0;
  background: transparent;
}

.c1 {
  width: 100%;
}

@media screen and (min-width:700px) {
  .c0 {
    padding: 0;
    margin: 0 6px 6px;
    max-width: 240px;
  }
}

<Comic
  id=""
  thumbnail={
    Object {
      "extension": "",
      "path": "",
    }
  }
  title=""
>
  <styled.button
    onClick={[Function]}
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      <ProgressiveImage
        placeholder="placeholder.jpg"
        src="."
      >
        <styled.img
          alt=""
          src="http://localhost/"
        >
          <img
            alt=""
            className="c1"
            src="http://localhost/"
          />
        </styled.img>
      </ProgressiveImage>
    </button>
  </styled.button>
</Comic>
`;
